### PR TITLE
Implement Adversarial Texture Attack on Object Trackers

### LINKED ISSUE #1326: Implement Physical Adversarial Textures
We would like to implement "Physical Adversarial Textures" that can be used as attack or defense against object trackers like GOTURN.

Paper link: https://arxiv.org/abs/1904.11042

### PR DESCRIPTION
**Pull request by beat-buesser**
# Description

Continuing from PR #1353

This pull request implements the Adversarial Texture attack on object trackers by Wiyatno and Xu, 2019 (https://arxiv.org/abs/1904.11042) in PyTorch compatible with ART's PyTorchGoturn object tracking estimator. This implementation supports segmentation masks of foreground objects to train adversarial textures that are robust against being partially covered by the tracked or other objects.

Fixes #1326

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [x] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/attacks/evasion/test_adversarial_texture_pytorch.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+@pytest.mark.skip_module("scripts")
+@pytest.mark.skip_framework("tensorflow", "keras", "kerastf", "mxnet", "non_dl_frameworks")
+def test_generate(art_warning, fix_get_mnist_subset, fix_get_goturn, framework):
+    try:
+        import torch
+        from scripts.train import GoturnTrain
+        from pathlib import Path
+
+        _device = "cpu"
```
Comment (GiulioZizzo) [inline tests/attacks/evasion/test_adversarial_texture_pytorch.py:51]: If a GPU is available, would it make sense to run the tests on a GPU to ensure that there is device consistency on this and future modifications of the code?

**Review by GiulioZizzo — COMMENTED** [review]
Hi Beat, many thanks for the review. I hope you find the comments helpful!

File: tests/attacks/evasion/test_adversarial_texture_pytorch.py (hunk 1 above)
Comment (beat-buesser) [reply tests/attacks/evasion/test_adversarial_texture_pytorch.py:51]: That's good idea and would be very nice, at the moment there are no GPUs available on Actions. At the moment we run the tests manually on a GPU.

File: art/attacks/evasion/adversarial_texture/adversarial_texture_pytorch.py [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+        :param videos: Video samples.
+        :param target: Target labels/boxes.
+        :param y_init: Initial labels/boxes.
+        :param foreground: Foreground mask.
+        :return: Loss.
+        """
+        import torch  # lgtm [py/repeated-import]
+
+        y_pred = self._predictions(videos, y_init, foreground)
+        loss = torch.nn.L1Loss(size_average=False)(y_pred[0]["boxes"].float(), target[0]["boxes"].float())
```
[the remark this answers is not in the document]
Comment (beat-buesser) [reply art/attacks/evasion/adversarial_texture/adversarial_texture_pytorch.py:195]: Good point, I'll replace it with `reduction='sum'`.
